src/neosophia/agents/react.py

```diff
@@ -44,33 +51,48 @@ def get_next_message(
 def make_react_agent(
         system_message: str,
         model: Callable,
-        function_descriptions: dict[str, Callable],
-        functions: list[dict],
-        max_llm_calls: int | None = 10
+        function_descriptions: list[dict],
+        functions: dict[str, Callable],
+        max_llm_calls: int | None = 10,
+        simple_formatting: bool | None = True
```

Comment:
The way these arguments are used you should remove the `| None`. You don't handle a case where they could take the value `None` so they aren't `Optional` in that sense. (`Optional[T]` means the variable could be `T` or `None`, it doesn't have anything to do with whether the argument has a default value.)

Reply:
Ah... good call out.